buffother.lic

```diff
@@ -0,0 +1,47 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#buffother
+=end
+custom_require.call(%w(common common-arcana events spellmonitor drinfomon common-travel))
+class Buffother
+  include DRC
+  include DRCA
+  include DRCT
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /\w+/, description: 'Person to buff' },
+        { name: 'waggle', regex: /\w+/, description: 'Waggle set to cast' },
+        { name: 'room', regex: /\d+/, optional: true, description: 'Room to move to' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+    @settings = get_settings
+    @recipient = args.recipient.capitalize
+    @waggle = args.waggle
+    @room = args.room
+    move_to_room
+  end
+
+  def move_to_room
+    echo "Moving to room"
+    DRCT.walk_to(@room)
+    cast_buff_spells
```

Comment:
To decouple the responsibilities of the methods, I recommend moving line 29 (cast_buff_spells) to just after line 23. In this way, the logic for moving to a room does not take on the logic of what is done while in that room. The higher order function `initialize` then orchestrates the two behaviors.

Reply:
Love it. Making that change.